shopping_mall/backend/ai/agent/holiday.py

```
@@ -0,0 +1,105 @@
+            items = [items]
+
+        holidays = set()
+        for item in items:
+            if item.get("isHoliday") == "Y":
+                locdate = str(item.get("locdate", ""))
+                if len(locdate) == 8:
+                    holidays.add(date(int(locdate[:4]), int(locdate[4:6]), int(locdate[6:])))
+
+        return holidays
+
+    except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)